src/single_file/c/illegal_instruction_experiments.c

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file    illegal_instruction_experiments.c
+ * @brief   Create illegal instruction exceptions for development of emulated instructions in ogsbi.
+ * 
+ * @copyright
+ *  Copyright (C) 2024 Eric Jessee\n
+ *  See the LICENSE file at the root of the project for licensing info.
+*/
+
+/* ------------------------------------------------------------------------------------------------
+ * Includes
+ * --------------------------------------------------------------------------------------------- */
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+/* ------------------------------------------------------------------------------------------------
+ * Function Implementations
+ * --------------------------------------------------------------------------------------------- */
+
+int main(void) {
+    puts("forcing a mulh instruction");
+    __asm__ volatile (
+        "li a7, -100\n"
+        "li a6, 10\n"
+        "mulh a5, a6, a7"
+    );
+    puts("forcing a mulhu instruction");
+    __asm__ volatile (
+        "li a7, 100\n"
+        "li a6, 55\n"
+        "mulhu a5, a6, a7"
+    );
+    puts("forcing a mulhsu instruction");
+    __asm__ volatile (
+        "li a7, -10\n"
+        "li a6, 100\n"
+        "mulhsu a5, a6, a7"
+    );
+    puts("Trying an unsigned divide (udiv)");
+    uint32_t a = 100;
+    uint32_t b = 10;
+    uint32_t c = a / b;
```

Comment:
`a`, `b`, and `c` need to all be declared `volatile` here, otherwise this check won't do anything in release builds (the division may be optimized out).

Same goes for the other non-assembly checks below.

Reply:
yep fair enough! That may explain why I was struggling to get the compiler to emit certain instructions the other day as well...